crates/iceberg/src/spec/manifest_list.rs

```diff
@@ -78,6 +78,11 @@ impl ManifestList {
     pub fn entries(&self) -> &[ManifestFile] {
         &self.entries
     }
+
+    /// Take ownership of the entries in the manifest list, consuming it
+    pub fn consume_entries(self) -> impl IntoIterator<Item = ManifestFile> {
```

Comment:
It's better to implement `IntoIterator` for `ManifestList` instead. This will allow users to use `ManifestList` more easily with other iterator-based APIs.

Reply:
Oh that's nicer. Will do in another PR now that this change was eliminated from this PR in the rebase.